components/module-components/blogs/interfaces.ts

```diff
@@ -0,0 +1,29 @@
+import { MetaData } from '../../../Types';
+
+export interface Author {
+  first_name: string;
+  last_name: string;
+}
+
+export interface Tag {
+  name: string;
+}
+
+export interface BlogPostAPI {
+  slug: string;
+  title: string;
+  tags: Tag[];
+  author: Author;
+}
+
```

Comment:
it seems like we have duplicate types, we have them here and also inside the queries